### DISCUSSION TITLE
Wie werden die Rollomotoren hinzugefügt?

### DISCUSSION THREAD

**Discussion by markus-hi** [dewenni/ESP32-Jarolift-Controller#14] [General]
Hallo,

ich hab das vermutlich nur überlesen, aber wie werden die Motoren in die Steuerung eingebunden? Muss ich diese dazu in den Anlernmodus setzen also auf die Programmiertaste am Motorkopf drücken?
Ich frage deswegen weil das bei mir etwas schwierig ist mit dem Zugang zu den Motoren....

Gruß - Markus

**Comment by flushbug** [comment]
Das ist über eine angelernte original Fernbedienung recht einfach. Auf+ab gleichzeitig drücken, danach 8 mal auf stop. Der Motor quittiert durch kurzes vibrieren den Anlernmodus. Nun kann er über die Web Oberfläche angelernt werden. War dies erfolgreich quittiert er erneut mit einem kurzen vibrieren. Fertig

**Reply by dewenni** [reply]
@flushbug danke für deine rege Beteiligung 👍

**Reply by ArkaOne** [reply]
> Das ist über eine angelernte original Fernbedienung recht einfach. Auf+ab gleichzeitig drücken, danach 8 mal auf stop. Der Motor quittiert durch kurzes vibrieren den Anlernmodus. Nun kann er über die Web Oberfläche angelernt werden. War dies erfolgreich quittiert er erneut mit einem kurzen vibrieren. Fertig

Bei mir funktioniert das nicht ! 🥲
Wenn ich nach auf+ab und 8 mal stop vom Motor quittiert bekomme, drücke ich im Web-GUI auf anlernen. Bekomme aber vom Motor keine Quittierung zurück. 
In der Log steht aber was vom anlernen. Ich kopiere es mal hier rein, vllt. kannst Du da was erkennen, wo der Fehler liegt !
Danke schon einmal...

[19.01.2025 - 03:20:37] D (APP-WEB): Received - Element ID: p12_learn_0 = true
[19.01.2025 - 03:20:37] D (APP-JARO): add single cmd to buffer: 5, 1
[19.01.2025 - 03:20:37] I (APP-WEB): cmd LEARN - channel 1
[19.01.2025 - 03:20:38] I (APP-JARO): putting channel 1 into learn mode ...
[19.01.2025 - 03:20:38] D (APP-JARO): get serial: 0x40776700 for channel: 1
[19.01.2025 - 03:20:38] D (APP-JARO): Device Counter from EEPROM: 39
[19.01.2025 - 03:20:38] D (APP-JARO): get serial: 0x40776700
[19.01.2025 - 03:20:38] D (APP-JARO): created devicekey low: 0xfe1f9da7 // high: 0xfe1f9da7
[19.01.2025 - 03:20:39] D (APP-JARO): Device Counter: 41
[19.01.2025 - 03:20:39] I (APP-JARO): Channel learned!
[19.01.2025 - 03:20:39] I (APP-JARO): execute cmd: LEARN - channel: 1
[19.01.2025 - 03:20:41] D (APP-WEB): Received - Element ID: p10_log_refresh_btn = true

Habe diese Bauteile in Verwendung !

![IMG_5696](https://github.com/user-attachments/assets/f455e59e-e0b8-43a0-9641-00f31ac5fb96)
![IMG_5697](https://github.com/user-attachments/assets/fa9a46bc-edb3-4ff7-a2be-6f60ae1b1f89)
![IMG_5698](https://github.com/user-attachments/assets/eed1a5b8-08f0-4b50-abb6-dc8ae14a3ae4)

LG
ArkaOne

**Reply by dewenni** [reply]
Die Bauteile sollten passen.
Hast du die Master Keys auch eingetragen?
Ansonsten teste nochmal mit einer anderen Seriennummer. Z.B. `000010` oder `010016` oder `0000ff`

Der Eintrag `I (APP-JARO): Channel learned!` zeigt leider nur das der Befehl ausgeführt wurde. Eine wirkliche Bestätigung vom Motor gibt es leider nicht.

**Reply by ArkaOne** [reply]
Ja, die Masterkeys habe ich eingetragen.
MasterMSB - 0xXXXX3a9b und MasterLSB - 0xXXXX0835
Die Seriennummern die Du genannt hast, habe ich versucht.
Leider immer mit dem gleichen negativen Ergebnis.

Ich habe mal irgendwo in einem Forum gelesen, das man es mit diesem Bauteil versuchen solle:

<img width="1089" alt="Bildschirmfoto 2025-01-19 um 14 54 11" src="https://github.com/user-attachments/assets/2a010204-b2cd-4f30-81d9-79a85c219eb6" />
 Habe mir diese mal bestellt zum ausprobieren.

Ich habe leider nur ein Multimeter zum Messen...
Wie könnte ich herausfinden, ob das 433 MHz Modul überhaupt funktioniert ?

LG
ArkaOne

**Reply by dewenni** [reply]
@ArkaOne 
zum testen könntest du auch den Log-Level mal auf "Debug" stellen und dann schauen ob du ein Signal von einer Fernbedienung empfängst.

Also hier umstellen auf "Debug"  und dann zur Sicherheit viellicht noch neu starten:
<img width="970" alt="image" src="https://github.com/user-attachments/assets/e1a13cb0-e6ed-4e24-82b1-2b003bc0c921" />

dann eine funktionierende Fernbedienung nehmen und in der Nähe des CC1101 mal ein paar Tasten drücken (Hoch, runter, etc)

Du solltest das dann im Log sehen:
```
[19.01.2025 - 19:26:49] D (APP-JARO): (INF3) rx_device_key_lsb: 0x8aea4711, rx_device_key_msb: 0xd40dc111, decoded: 0x01003d11
[19.01.2025 - 19:26:49] D (APP-JARO): (INF2) RSSI: 80, counter: 185
[19.01.2025 - 19:26:49] D (APP-JARO): CC1101_RSSI: 80
[19.01.2025 - 19:26:49] D (APP-JARO): (INF1) serial: 0x0c7c0000, rx_function: 0x2, rx_disc_low: 1, rx_disc_high: 72
```

Wenn du da nichts empfängst, dann passt denke ich noch was mit dem CC1101 nicht.

**Reply by ArkaOne** [reply]
So, Problem gelöst...

Die Antennenbuchse hatte eine kalte Lötstelle... und das bei Neu gekauft. Man,man,man

Danke für den Tip !
Das mit dem Log hat mich erst darauf gebracht, mal die Lötstellen des Moduls zu checken..

Nun funktionieren die Rolladen bestens ! ;)

LG
ArkaOne

**Reply by dewenni** [reply]
super, danke für die Rückmeldung und viel Spaß damit!

**Comment by markus-hi** [comment]
Ah super! Dann ist das wirklich einfach und ich mach mich mal gleich ans basteln 👏🙋‍♂️

Vielen Dank!

**Reply by dewenni** [reply]
viel Erfolg - schon witzig, du nutzt auch mein Buderus Projekt, oder?

**Reply by markus-hi** [reply]
Ja korrekt, das Buderus Projekt nutze ich auch. Das läuft schon lange, wollte ich eigentlich über die Tage mal updaten. Aber das mit den Rollomotoren hat jetzt erstmal Prio 👌😉🤣
Danke, auch für deine ganze Zeit die du hier diese Projekte investierst 👏

**Comment by ben-ehlers** [comment]
Moin, leider klappt es bei mir einfach nicht also der Motor vibriert und ich drücke auch auf den Knopf anlernen, welche Nummern muss ich bei Seriennummer usw. eingeben ? Es klappt einfach nicht, habt ihr Tipps oder wo der Fehler liegen könnte ?

**Reply by detti77** [reply]
Hast du den Anlernmodus über die Fernbedienung oder den Taster am Motor aktiviert? Bei mir hat das über die Fernbedienung nicht geklappt. Über den Taster am Motor hat es dann sofort funktioniert.
Bei der Serial habe ich einfach die genommen, die ich aus der vorhandenen Fernbedienung (Logbuch) ausgelesen habe und dann in den letzten Bytes leicht verändert habe. Bitte nicht die aus der FB nehmen. Dann laufen die irgendwann auseinander und di musst auch die FB wieder anlernen.

**Reply by dewenni** [reply]
Hi @ben-ehlers 
Hast du MSB und LSB eingetragen?

Siehe auch hier:
https://github.com/dewenni/ESP32-Jarolift-Controller/issues/37#issuecomment-2692673536

**Reply by ziegenpaul99** [reply]
@dewenni ja habe ich eingegeben allerdings klappt es nicht ich sehe im log die Daten der Fernbedienung aber es klappt einfach nicht

**Reply by dewenni** [reply]
@ziegenpaul99 
also wenn du MSB und LSB korrekt eingetragen hast und auch als Seriennummer mal die 000010 probiert hast, dann sollte das schon funktionieren.
Scheitert das Anlernen derzeit bei dir?
Versuchst du das Anlernen über den Knopf am Motor oder durch kopieren einer Fernbedienung?
Versuche hier auch sicherheitshalber mal das Anlernen durch drücken der Taste am Motor.
Ansonsten wäre noch die Frage wie alt deine Motoren sind. Bei älteren Motoren muss man wohl auch den "alten Anlernmodus" nutzen. Dazu gibt es ja auch eine Einstellung. Das konnte ich aber nicht testen, weil ich nur neue Motoren habe.
